kotest-assertions/src/jvmMain/kotlin/io/kotest/Retry.kt

```diff
@@ -0,0 +1,58 @@
+package io.kotest
+
+import io.kotest.assertions.Failures
+import kotlinx.coroutines.delay
+import java.time.Duration
+
+/**
+ * Retry the given lambda [f] in case of assertion error[AssertionError] and exception [Exception].
+ * Stops trying when [maxRetry] or [maxDelay] duration is reached.
+ * */
+suspend inline fun <T> retry(
+   maxRetry: Int,
+   maxDelay: Duration,
+   delay: Duration = maxDelay,
+   multiplier: Int = 1,
+   f: () -> T
+): T = retry(maxRetry, maxDelay, delay, multiplier, Exception::class.java, f)
```

Comment:
What do you think about extracting the maxRetry/maxDelay/delay logic out into an interface `Retry` of which one could be `SimpleRetry` which implements this.

It means we could add `BackoffRetry` later as well.

Or do you think that's overkill ?

Reply:
I am not sure how much advantage will extracting them in a Interface will provide. Which is based on my assumption this would look like after extracting in Interface.

```kotlin
interface RetryStrategy {
   val maxRetry: Int
   fun maxDelay(): Duration
   fun delay(): Duration
}

class SimpleRetryStrategy(override val maxRetry: Int, private val maxDelay: Duration, private val delay: Duration) :
   RetryStrategy {
   override fun maxDelay(): Duration = maxDelay
   override fun delay(): Duration = delay
}

class BackoffStrategy(override val maxRetry: Int, private val maxDelay: Duration, private val delay: Duration) :
   RetryStrategy {
   var counter = 1L
   override fun maxDelay(): Duration = maxDelay
   override fun delay(): Duration {
      val nextDelay = delay.multipliedBy(counter)
      counter += 1L
      return nextDelay
   }
}
```

Reply:
Ok I think leave it as it was. But one small change can we rename delay to timeout.

Reply:
Sorry to be clear delay is fine but maxDelay I think is better as timeout.

Reply:
Done